[Arnav Aggarwal] ip

arnav-ag opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/arnav-ag/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE!__

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    Duke duke = new Duke(DIRECTORY, FILENAME);
    duke.run();
}
```

arnav-ag committed:
Level-1

arnav-ag committed:
Level-2

arnav-ag committed:
Level-3

arnav-ag committed:
Level-4

arnav-ag committed:
A-TextUiTesting

arnav-ag committed:
Level-5

arnav-ag committed:
Level-6

arnav-ag committed:
Add save to file and load from file feature

arnav-ag committed:
Refactor code to follow coding standards

arnav-ag committed:
Refactor code to introduce more methods

arnav-ag committed:
Add date parsing feature for deadline and events

arnav-ag committed:
Merge branch 'branch-Level-7'

arnav-ag committed:
Merge branch 'branch-Level-8'

arnav-ag committed:
Refactor code to introduce more classes

arnav-ag committed:
Refactor code to introduce packages

arnav-ag committed:
Change tests to account for packages

arnav-ag committed:
Add various exceptions for better verbosity

arnav-ag committed:
Add gradle build files

arnav-ag committed:
Add parser package separately

arnav-ag committed:
Add parser unit tests

arnav-ag committed:
Add JavaDoc

arnav-ag committed:
Refactor code to follow coding standards

arnav-ag committed:
Refactor code to follow coding standards

arnav-ag committed:
Add find feature to Todo list

arnav-ag committed:
Merge branch 'branch-A-CodingStandard'

arnav-ag committed:
Add JavaDoc

arnav-ag committed:
Merge branch 'branch-A-JavaDoc'

arnav-ag committed:
Merge branch 'branch-Level-9'

arnav-ag committed:
Modify imports to follow coding standards

arnav-ag committed:
Refactor code to prevent unnecessary System calls

arnav-ag committed:
Add checkstyle and refactor code and test accordingly

arnav-ag committed:
Add checkstyle and refactor code and test accordingly

arnav-ag committed:
Refactor code

arnav-ag committed:
Refactor code

arnav-ag committed:
Refactor code to remove unnecessary newlines

arnav-ag committed:
Merge branch 'branch-A-CheckStyle'

MuhdMusab commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+    private static String getSecondHalf(String command) {
```
Perhaps the method name could be a bit more descriptive?

MuhdMusab commented on src/main/java/duke/command/ModifyCommand.java:
```diff
@@ -0,0 +1,65 @@
+    /**
+     * Types of commands that can be executed by this class.
+     */
+    public enum CommandType {
```
Should all subclasses of command have a CommandType attribute similar to this ModifyCommand class? Or perhaps List, Undone and Done commands should be their own separate Commands?

MuhdMusab commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+            }
+            reader.close();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Should the user be allowed to view the stack trace when an error occurs?

MuhdMusab reviewed (commented):
Good use of SLAP coding principle. Very easy to read and understand.
Some cosmetic changes requested.

XenonShawn commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Generic Command parent class.
+ */
+public class Command {
```
Consider making this an abstract base class instead

XenonShawn commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+                currString += currTask.isDone() ? "1 | " : "0 | ";
+                currString += currTask.getDescription() + " | ";
+                currString += currTask.getDate();
+            }
```
Could this be done better with polymorphism? Perhaps it would be cleaner if you invoked a method of the `Task` class instead

XenonShawn reviewed (commented):
Overall LGTM, good job on keeping your code clean and easy to read :)

arnav-ag committed:
Introduce GUI to app

arnav-ag committed:
Resolve checkstyle errors

arnav-ag committed:
Merge branch 'branch-Level-10'

arnav-ag committed:
Resolve unnecessary use of fill

arnav-ag committed:
Merge branch 'branch-Level-10'

arnav-ag committed:
Add assertion to MainWindow

MainWindow checks for and asserts that the user input is not empty.
This is necessary as previously when a user inputted an empty string,
the message will be "displayed" as a non-existent dialog box with an
invalid command error from the bot.

arnav-ag committed:
Made changes to follow checkstyle

Duke.java: Add javadoc for constructor
ExitCommand.java: Add javadoc for execute method
DialogBox.java: Remove unecessary imports
Launcher.java: Add javadoc for class
Parser.java: Add/remove newlines for readlability

arnav-ag committed:
Follow code quality convention

arnav-ag committed:
Merge pull request #3 from arnav-ag/branch-A-Assertions

Add assertion to MainWindow

arnav-ag committed:
Merge pull request #2 from arnav-ag/branch-A-CodeQuality

Follow code quality convention

arnav-ag committed:
Add check for duplicate task

arnav-ag committed:
Add user guide documentation

arnav-ag committed:
Make changes to response handling

arnav-ag committed:
Refactor code to follow coding standards

arnav-ag committed:
Edit docs to include link to jar file
